scripts/GetDependentBranchNames/Program.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Medallion.Shell;
+
+namespace BuildNugetPackages
+{
+
+    internal class Program
+    {
+        private static readonly Shell shell = new Shell(options => options.ThrowOnError());
+
+        private struct Dependency
+        {
+            public string Label { get; set; }
+            public string Repo { get; set; }
+            public string Trigger { get; set; }
+        }
+
+        private static readonly List<Dependency> Dependencies = new List<Dependency>
+        {
+            new Dependency
+            {
+                Label = "build-database-sync-worker",
+                Trigger = "database-sync-worker-premerge",
+                Repo = "https://github.com/spatialos/database-sync-worker.git"
+            }
+        };
+
+        private static int Main(string[] args)
+        {
+            try
+            {
+                var currentBranch = Environment.GetEnvironmentVariable("BUILDKITE_BRANCH");
+
+                if (string.IsNullOrEmpty(currentBranch))
+                {
+                    throw new Exception("The BUILDKITE_BRANCH environment variable is empty or not set.");
+                }
+
+                foreach (var dep in Dependencies)
+                {
+                    var lines = new List<string>();
+                    shell.Run("git", "ls-remote", "--heads", dep.Repo, currentBranch)
+                        .RedirectTo(lines)
+                        .RedirectStandardErrorTo(Console.Error)
+                        .Wait();
+
+                    var remoteBranch = lines.Any() && lines.First().Contains(currentBranch) ? currentBranch : "master";
```

Comment:
`lines.Any()` could be not-needed here? 

```csharp
var remoteBranch = lines.FirstOrDefault()?.Contains(currentBranch) ? currentBranch : "master";
```

(I'm unsure if C# can coerce the `bool?` to `bool` here)

Reply:
I wrote it like that originally, but didn't like it for whatever reason